pkg/orchestrator/provider/kubernetes/provider.go

```diff
@@ -113,10 +120,215 @@ func (p *Provider) DiscoverAssets(ctx context.Context) provider.AssetDiscoverer
 	return assetDiscoverer
 }
 
-func (p *Provider) RunAssetScan(context.Context, *provider.ScanJobConfig) error {
-	return fmt.Errorf("not implemented")
+// nolint: cyclop
+func (p *Provider) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+	discoverers, err := p.clientSet.CoreV1().Pods(p.config.ContainerRuntimeDiscoveryNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.Set(crDiscovererLabels).String(),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to list discoverers: %w", err)
+	}
+
+	objectType, err := config.AssetInfo.ValueByDiscriminator()
+	if err != nil {
+		return fmt.Errorf("failed to get asset object type: %w", err)
+	}
+
+	clients := []*containerruntimediscovery.Client{}
+	for _, discoverer := range discoverers.Items {
+		discovererEndpoint := net.JoinHostPort(discoverer.Status.PodIP, "8080")
+		clients = append(clients, containerruntimediscovery.NewClient(discovererEndpoint))
+	}
+
+	switch value := objectType.(type) {
+	case models.ContainerImageInfo:
+		var pickedClient *containerruntimediscovery.Client
+		for _, client := range clients {
+			_, err := client.GetImage(ctx, value.ImageID)
+			if err == nil {
+				pickedClient = client
+				break
+			}
+		}
```

Comment:
should we parallelize this? is it okay to not log anything in case all ops fail?

Reply:
we could also move it to a separate `selectClient(clients []client, selector func(client) error) (client, error)` function which accepts functor that performs an operation (e.g. `GetImage`, `GetContainer`) and returns a client on success so that we can reuse it for other ops

Reply:
I am okay with the other notes. But we do ignore the context of why the other clients failed since we return a generic error (unable to find image ID). it can happen that the id exists, but there is some other issue with the client. maybe aggregate all the errors together and only log them in case no client can be selected. anyhow, I'm okay with this.